core/logic_events.go

```diff
@@ -668,6 +684,167 @@ func contactsToDef(items []model.ContactLegacy) []model.ContactLegacy {
 	return defs
 }
 
+func counstructImage(originatingCalendarID, dataSourceEventID, eventID string, prefixPath string) string {
```

Comment:
We have discussed that you have to make all images processing before to apply the transaction as currently it will fail for sure with time out. You have to prepare for example 1000 or 2000 images which emans 1000 and 2000 time to ask webtools and then to upload to content bb.

Also, it will be well done if you introduce Images driven adapter and move there all logic related with the images.